ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -659,9 +662,15 @@ public void shouldDistributeMultipleInterDependantStatements() {
     // Then:
     final InOrder inOrder = inOrder(commandStore);
     inOrder.verify(commandStore).enqueueCommand(
-        eq("CREATE STREAM S AS SELECT * FROM test_stream;"), any(), any(), any());
+        eq("CREATE STREAM S WITH (REPLICAS = 1, PARTITIONS = 4, KAFKA_TOPIC = 'S') "
+               + "AS SELECT TEST_STREAM.S2_F1 \"S2_F1\"\n"
+               + "FROM TEST_STREAM TEST_STREAM;"), any(), any(), any());
     inOrder.verify(commandStore).enqueueCommand(
-        eq("CREATE STREAM S2 AS SELECT * FROM S;"), any(), any(), any());
+        eq("CREATE STREAM S2 WITH (REPLICAS = 1, PARTITIONS = 4, KAFKA_TOPIC = 'S2') AS SELECT\n"
+               + "  S.ROWTIME \"ROWTIME\"\n"
```

Comment:
The current behaviour of

```sql
SELECT ROWKEY from Blah;
```

Would be to _copy_ the rowkey into the value.  Same for ROWTIME. So this inferred projection schema is wrong.

Reply:
Is this a bug in the SQL formatter? At first I was a little surprised by it as well, but this was just the output of formatting the statement.

Reply:
I wanted to put some thoughts I had here (summarizing discussion with @mjsax): To align with the conclusion of #824 we really need to think about what `*` really means. Consider the three statements below.
```sql
1: SELECT * FROM foo;
2: CREATE STREAM bar AS SELECT * FROM foo;
3: SELECT * FROM bar;
```
There is something counterintuitive going on here, especially with regards to `ROWTIME`: the results of `1` and `3` can differ! Well of course... that's not particularly surprising, but it does beg a question: is `ROWTIME` really part of `*`?

I think the answer to that should be no (as you suggest, that the inferred projection is wrong). I think this means that we need to do one of the following:

- Change the format of our transient query from `ROWTIME | ROWKEY | V1 | V2 ...` to `V1 | V2 ... (key: KEY, ts: ROWTIME)`
- Require `SELECT ROWKEY, ROWTIME, *` (which also copies them into the value)

I prefer the first alternative, which makes it clear that (today, this will change when we implement structured keys) `ROWTIME` and `ROWKEY` are not part of the attributes. Today the code is pretty unclear about it, having ad hoc calls to `SchemaUtil#add/removeImplicitFields` - we should make it clear and consistent, pushing these operations only to the leaves of the execution trees and never storing them in the schema.

Reply:
I tried to implement this, but it looks like the implicit fields are extremely pervasive throughout not only code but also data - we copy these fields in some scenarios and not necessarily others and changing this behavior would be quite tricky. In the meantime, I've opened #2473 to address this issue and will put this PR on hold until that is checked in.

Reply:
Agreed the code is a mess in this area. (I'm hoping to get a chance to clean some of it up with the structured keys work).

Not sure what you mean by:

> * Change the format of our transient query from `ROWTIME | ROWKEY | V1 | V2 ...` to `V1 | V2 ... (key: KEY, ts: ROWTIME)`

Can you elaborate? If you're suggesting we change the output of a query in the CLI, then personally, I don't see the need / point.

Regarding fixing SqlFormatter - yes, that should be done.  Be careful though, as it's used, (as you've seen), when baking details into statements before writing to the command topic.  _Because_ its used in a such a critical path in the system we should probably ensure it works! :D

Reply:
After thinking about it for a while, my suggestion boils down to two things:

- `SELECT *` should not select `ROWTIME` or `ROWKEY` (as it stands today)
- It is useful, though, for the CLI to display the time and key of each row in a `SELECT *`. In order to make the semantics clear to the user, my suggestion is to put them in parentheses and label them explicitly instead of lumping them into the values returned. 

Either way, I figured that doing the above is a project in and of itself (and keeping everything else working) so I went with the other solution.

Reply:
> my suggestion is to put them in parentheses and label them explicitly instead of lumping them into the values returned.

I agree for the current data model. When we switch to the logical model from our structured keys design discussions, I think it does make sense to include the key (not the rowtime) as part of the main schema in select results.

Reply:
I'd suggest leaving this alone in the context of this PR and addressing it as part of the structured keys work

Reply:
Yup - the only thing I did for this PR was maintain the `SELECT *` functionality.